### PR TITLE
add geosnap

### PR DESCRIPTION
**Pull request by knaaptime**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by knaaptime** [comment]
@sjsrey would you mind commenting here?

**Comment by sjsrey** [comment]
I am willing to be a co-maintainer.

**Comment by knaaptime** [comment]
tests are passing on linux and macos but windows seems to be hitting some kind of encoding error?

**Comment by knaaptime** [comment]
now i've got all platforms passing but appveyor failing?

**Comment by knaaptime** [comment]
I'm pretty confident this should be good to go

**Comment by knaaptime** [comment]
@conda-forge-admin please rerender

**Comment by knaaptime** [comment]
i figured that probably wouldnt work while this is a PR, but it was worth a shot

File: recipes/geosnap/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - pip
+  run:
+    - python >=3
+    - pandas
+    - giddy >=2.2.1
+    - libpysal
+    - spenc
+    - cenpy
+    - geopandas
+    - matplotlib
```
Comment (scopatz) [inline recipes/geosnap/meta.yaml]: Can this be `matplotlib-base`, or does it need the Qt backend?

File: recipes/geosnap/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    dynamics. It provides a suite of tools for creating socio-spatial datasets, harmonizing those 
+    datasets into consistent set of time-static boundaries, and modeling neighborhood change using 
+    classic and spatial statistical methods.
+  doc_url: http://spatialucr.github.io/geosnap
+  dev_url: https://github.com/spatialucr/geosnap
+
+extra:
+  recipe-maintainers:
+    - knaaptime
+    - sjsrey
```
Comment (scopatz) [inline recipes/geosnap/meta.yaml]: @sjsrey - are you OK with being a maintainer?

**Comment by scopatz** [comment]
Appveyor is allowed to fail on staged-recipes, if it is a noarch package or the azure windows build passes

File: recipes/geosnap/meta.yaml (hunk 2 above)
Comment (sjsrey) [reply recipes/geosnap/meta.yaml]: Yes on being a maintainer.

**Pull request merged**